tasks.py

```diff
@@ -782,14 +785,72 @@ def show_automerges(ctx: Context):
         print("\tNo PRs set to automerge")
 
 
-MARKER_REQ_MAPPING: Final[Mapping[str, tuple[str, ...]]] = {
-    "athena": ("reqs/requirements-dev-athena.txt",),
-    "cloud": ("reqs/requirements-dev-cloud.txt",),
-    "pyarrow": ("reqs/requirements-dev-arrow.txt",),
-    "external_sqldialect": ("reqs/requirements-dev-sqlalchemy.txt",),
+class TestDependencies(NamedTuple):
+    requirement_files: tuple[str, ...]
+    services: tuple[str, ...] = tuple()
+    setup_funcs: tuple[Callable[[Context], Result | None], ...] = tuple()
+    exta_pytest_args: tuple[  # TODO: remove this once remove the custom flagging system
+        str, ...
+    ] = tuple()
+    # NOTE: We probably don't need both `services` and `setup_funcs`
+    # but it's not clear to me which one is better
+    # `services` is simpler because it's just a list of services to up with docker-compose
+    # but `setup_funcs` is more flexible because it allows us to do more complex setup
+    # It could do the same docker-compose up or anything else.
+    # setup_funcs is a tuple of functions that take a an `invoke.Context` and return an invoke `Result` or `None`
+    # The invoke context can be used to run CLI subcommands or can be ignored.
+    # More arguments could be added to this `Callable` interface if needed.
+
+
+MARKER_DEPENDENDENCY_MAP: Final[Mapping[str, TestDependencies]] = {
+    "athena": TestDependencies(("reqs/requirements-dev-athena.txt",)),
+    "cloud": TestDependencies(
+        ("reqs/requirements-dev-cloud.txt",), exta_pytest_args=("--cloud",)
+    ),
+    "external_sqldialect": TestDependencies(("reqs/requirements-dev-sqlalchemy.txt",)),
+    "pyarrow": TestDependencies(("reqs/requirements-dev-arrow.txt",)),
+    "postgresql": TestDependencies(
+        ("reqs/requirements-dev-postgresql.txt",),
+        setup_funcs=(
+            lambda ctx: ctx.run(
+                "docker-compose -f assets/docker/postgresql/docker-compose.yml up -d"
+            ),
+        ),
+        exta_pytest_args=("--postgresql",),
+    ),
+    "spark": TestDependencies(
+        ("reqs/requirements-dev-spark.txt",),
+        # services=("spark",),
+        exta_pytest_args=("--spark",),
+    ),
 }
 
 
+# TODO: return list & lru_cache this??
+def _tokenize_marker_string(marker_string: str) -> Generator[str, None, None]:
+    """
+    Split a marker string on ' or ' and ' and ' and return an iterable of tokens.
+    Exclude parts of the string after ' and not '.
+    """
+    remaining_str, _, _ = marker_string.partition(" and not ")
```

Comment:
This feels fragile to me. We're basically building a parser for pytests marker syntax and we've made assumptions that will bite use later if we think we're using the full parser. Right now this only supports one `not`, it must be at the end, and it must be preceded by another marker. 

This makes it coupled to how we call tests in the `ci.yml` file even though it looks generic.

Alternatives:
* Use the parser from pytest. I looked at this a little bit and, at first glance, it looks like the marker string parsing is private. We could dig a little more.
* Explicitly handle any markers that have more than 1 token in them. That is, have explicit cases for `openpyxl or pyarrow or project or sqlite` and `cloud and not e2e`. With this we can throw an clear error message if we try to run this with a combo we haven't seen before.

Reply:
I'm totally with you. I think the pytest parser would be the "correct" way to go, and to "hardcode" for now.